### DISCUSSION TITLE
PrismaClient.update maps to what function in MongoDb?

### DISCUSSION THREAD

**Discussion by YesYouKenSpace** [prisma/prisma#13985] [Q&A]
Hi everyone, 

I tried tracing the code but since I don't know Rust it is a little difficult for me to understand.

I want to know if `PrismaClient.update` is an atomic filter + update like MongoDb's [findAndModify](https://www.mongodb.com/docs/manual/reference/method/db.collection.findAndModify/)

My tracing got me here https://github1s.com/prisma/prisma-engines/blob/HEAD/query-engine/connectors/mongodb-query-connector/src/root_queries/write.rs#L122 which seems to suggest that it does not as there is only `update_records` and not `update_record`

Related:
1. https://github.com/prisma/prisma/issues/8612 - but Postgres and uses `updateMany`

**Comment by austincrim** [comment]
Hey @kennethtxytqw 👋,

It looks like Prisma generates a MongoDB [`updateMany`](https://www.mongodb.com/docs/manual/reference/method/db.collection.updateMany/) query when using the `prisma.update` method.

You can always turn on [query logging](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging#log-to-stdout) to see what kind of queries the client is generating.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
